### DISCUSSION TITLE
Migrating from ipsec-tools, host-to-host IPSec tunnel

### DISCUSSION THREAD

**Discussion by MegaBrutal** [strongswan/strongswan#791] [Q&A] [answered]
I've been using ipsec-tools previously, but this package got removed from Ubuntu and turns out it's not maintained anymore, so using it today is not a good idea. I've read Strongswan might be a viable replacement, but its configuration is vastly different and it's hard to wrap my head around it.

My use case is very simple: I want 2 hosts to communicate over an encrypted IPSec tunnel to protect an NFS mount. That's it, no VPN or other fancy stuff.

In the ipsec-tools era, I followed this guide to set up such tunnel:
https://help.ubuntu.com/community/IPSecHowTo

How it works: whenever host A wants to send a packet to host B, it gets encrypted and encapsulated in an AH/ESP packet. Both hosts only accept packets from the other if properly encrypted with the right PSK. Unencrypted packets and packets encrypted with the wrong key are silently dropped. On the firewall I need to set that I only allow AH/ESP communication between the hosts; raw TCP, UDP, ICMPv6 packets are dropped. The hosts can't even ping each other with unencrypted ICMPv6 Echo requests and replies; indeed when I run ping6, I only see AH/ESP packets exchanged with tcpdump. I realize that it might be the most basic utilization of IPSec while the protocol is capable of securing solid site-to-site VPNs; but this is all I need for my use case.

How can I get Strongswan to behave the same or similarly? I realize Strongswan might insist to do IKEv2 negotiations to exchange keys and not just encrypt with a predefined PSK like the above ipsec-tools example – that is fine, even better. But I certainly want to keep the functionality that the hosts should never ever be allowed to exchange unencrypted packets, especially packets related to the NFS mount.

I was browsing the examples, but haven't found one that's quite like the use case I have; so I tried to use 2 others as a template.

This is what I currently have on one host, guess the remote host should be the mirror of this (obviously I censored hostnames and IP addresses, and the key is different):

```
connections {

	remotehost {
		local_addrs = XXXX::2001
		remote_addrs = YYYY::2002

		local {
			auth = psk
			id = local.domain
		}
		remote {
			auth = psk
			id = remote.domain
		}

		children {
			remotehost {
				local_ts = XXXX::2001
				remote_ts = YYYY::2002
				ah_proposals = aesxcbc
				mode = transport
			}
		}

		version = 2
		mobike = no
		proposals = aes128-aesxcbc-modp3072

	}
}

secrets {

	ppk-remotehost {
		id = remote.domain
		secret = 0xc3b62e215cf44b7d5dab0874c7fd2adf8a52d2c15ce0fbea
	}

}
```

What happens:
I reload the config with ``swanctl --load-all``.

Then ``swanctl --list-conns`` gives the following:
```
remotehost: IKEv2, no reauthentication, rekeying every 14400s
  local:  XXXX::2001
  remote: YYYY::2002
  local pre-shared key authentication:
    id: local.domain
  remote pre-shared key authentication:
    id: remote.domain
  remotehost: TRANSPORT, rekeying every 3600s
    local:  XXXX::2001/128
    remote: YYYY::2002/128
```

But ``swanctl --list-sas`` and ``swanctl --list-pols`` gives nothing! When I ping the other host, they exchange unencrypted ICMPv6 packets... so it doesn't work at all.

Strongswan is really new for me and honestly I was hoping to find a more direct replacement for ipsec-tools; sure I'm missing some fundamental concepts in the configuration. Can you please light me up what I might be doing wrong?

**Comment by tobiasbrunner** [comment, marked as answer]
> Strongswan is really new for me and honestly I was hoping to find a more direct replacement for ipsec-tools; sure I'm missing some fundamental concepts in the configuration.

Please see [IntroductionTostrongSwan](https://wiki.strongswan.org/projects/strongswan/wiki/IntroductionTostrongSwan). Anyway, what you want is to configure `start_action=trap`  (see the section called "Invocation and Maintenance" in the above document and the [swanctl.conf reference](https://wiki.strongswan.org/projects/strongswan/wiki/swanctlconf) for details).
